### DISCUSSION TITLE
RAG fails when documents increase

### DISCUSSION THREAD

**Discussion by snadeem1362** [open-webui/open-webui#3995] [General]
# Bug Report
RAG fails for large number of documents
## Description
We have a huge number of pdf documents. We want the whole data to be ingested in RAG.
We are able to ingest data and generate embeddings, although it took a month to generate embeddings for 70GB PDF docs. The final vector DB size is 220GB. Now when we try to ask questions using these documents as context it just never returns and keep on loading. 
Is Open-webui RAG suitable for such huge amount of data? What are the alternatives?
**Bug Summary:**
Open-webui RAG just cant answer when embedding database is huge like 200GB

**Steps to Reproduce:**
Upload 20GB of documents in scan directory
Generate embeddings using default all-mini-llm-v2 embedding model
Set these documents as context
Ask question
It will never return and keep on loading

**Expected Behavior:**
It should come up with a precise answer from the embedded documents

**Actual Behavior:**
It is unable to answer and keep on searching 

## Environment

- **Open WebUI Version:** [3.7]
- **Ollama (if applicable):** [0.1.30]

- **Operating System:** [Ubuntu 22.04]
- **Browser (if applicable):** [Firefox 98.0]

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
It just keep in searching relevant chunks from vector DB

## Installation Method

Docker GPU Version

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by rolandomar** [comment]
Have the same issue with around 2500 pdfs (1.6GB) resulting im 40GB of embeddings.

**Reply by snadeem1362** [reply]
So what is the solution? Are you able to tackle this somehow?

**Reply by leandropjp** [reply]
Any update?

**Comment by Tintalala** [comment]
Is it because chromadb is using sqllite?  can postgres pgvector be used for embedding?
